File: 007.跨端APP应用（移动端开发工程师）/tests/finance.test.ts
import { describe, it, expect } from 'vitest'
import { formatAmount, typeOfAccount, typeOfCategory, balanceSign } from '@/utils/finance'

describe('finance utils', () => {
  it('formatAmount: 正数带 ¥,两位小数', () => {
    expect(formatAmount(12.5)).toMatch(/12\.50/)
    expect(formatAmount(0)).toMatch(/0\.00/)
  })
  it('typeOfAccount', () => {
    expect(typeOfAccount('cash')).toBe('现金')
    expect(typeOfAccount('debit')).toBe('借记卡')
  })
  it('typeOfCategory: expense 走支出图标, income 走收入图标', () => {
    expect(typeOfCategory('expense')).toBe('expense')
    expect(typeOfCategory('income')).toBe('income')
  })
  it('balanceSign', () => {
    expect(balanceSign(100)).toBe(1)
    expect(balanceSign(-50)).toBe(-1)
    expect(balanceSign(0)).toBe(0)
  })
})

File: 007.跨端APP应用（移动端开发工程师）/flutter-project/test/unit/finance_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:qingzhang_app/core/api/models.dart';
import 'package:qingzhang_app/core/utils/finance.dart';

void main() {
  group('formatAmount', () {
    test('formats with two decimals', () {
      expect(formatAmount(12.34), '12.34');
    });

    test('zero', () {
      expect(formatAmount(0), '0.00');
    });

    test('withSymbol adds ¥', () {
      expect(formatAmount(1.5, withSymbol: true), startsWith('¥'));
    });
  });

  group('typeOfAccount', () {
    test('cash returns non-empty', () {
      expect(typeOfAccount(AccountType.cash), isNotEmpty);
    });
  });

  group('balanceSign', () {
    test('negative returns -1', () {
      expect(balanceSign(-100), -1);
    });
    test('positive returns 1', () {
      expect(balanceSign(100), 1);
    });
    test('zero returns 0', () {
      expect(balanceSign(0), 0);
    });
  });
}